File: c/darts/src/darts.c
#include "darts.h"
#include <math.h>

int score(coordinate_t pt) {
    const double radius = sqrt(pt.x * pt.x + pt.y * pt.y);

    if (radius > 10) return 0;
    else if (radius > 5) return 1;
    else if (radius > 1) return 5;
    
    return 10;
}

File: ruby/darts/darts.rb
class Darts
  def initialize(x, y)
    @x = x
    @y = y
  end

  def score()
    r = Math.sqrt(@x**2 + @y**2)
    return 10 if r <= 1
    return 5 if r <= 5
    return 1 if r <= 10
    0
  end
end